Patch-Source: https://gitlab.com/samba-team/samba/-/merge_requests/3981
(rebased)
---
From 0b3cb978cc6d4a66891aead7edd35110725083e5 Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Mon, 3 Mar 2025 11:22:12 +0100
Subject: [PATCH] lib: Fix missing include <sys/types.h>

uid_t is provided by sys/types.h according to POSIX

Building sssd (https://github.com/SSSD/sssd) with --with-samba and musl libc
results in:

	/usr/include/samba-4.0/ndr.h:797:25: error: unknown type name 'uid_t'; did you mean 'pid_t'?
	  797 | NDR_SCALAR_PROTO(uid_t, uid_t)
	      |                         ^~~~~
	...
	/usr/include/samba-4.0/gen_ndr/security.h:639:9: error: unknown type name 'uid_t'; did you mean 'pid_t'?
	  639 |         uid_t uid;
	      |         ^~~~~
	      |         pid_t

Ref: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html#tag_13_65
Signed-off-by: Achill Gilgenast <fossdd@pwned.life>
Signed-off-by: fossdd <fossdd@pwned.life>
---
 libcli/security/security.h | 2 ++
 librpc/ndr/libndr.h        | 1 +
 2 files changed, 3 insertions(+)

diff --git a/libcli/security/security.h b/libcli/security/security.h
index a1c26ed57f47..6d24d3702305 100644
--- a/libcli/security/security.h
+++ b/libcli/security/security.h
@@ -25,6 +25,8 @@
 
 #include "librpc/gen_ndr/security.h"
 
+#include <sys/types.h>
+
 /* File Specific access rights */
 #define FILE_READ_DATA        SEC_FILE_READ_DATA
 #define FILE_WRITE_DATA       SEC_FILE_WRITE_DATA
diff --git a/librpc/ndr/libndr.h b/librpc/ndr/libndr.h
index 344f08b20a1c..a245fb74b2f6 100644
--- a/librpc/ndr/libndr.h
+++ b/librpc/ndr/libndr.h
@@ -26,6 +26,7 @@
 #define __LIBNDR_H__
 
 #include <talloc.h>
+#include <sys/types.h>
 #include "../lib/util/discard.h" /* for discard_const */
 #include "../lib/util/data_blob.h"
 #include "../lib/util/time.h"
-- 
GitLab

